### PR TITLE
chore_: use status.go v2 endpoint

### PR DESCRIPTION
**Pull request by qfrank**
this is a followup PR for [PR 21312](https://github.com/status-im/status-mobile/pull/21312#issuecomment-2376621131)

related status-go [PR](https://github.com/status-im/status-go/pull/5943)

new endpoints updated in this PR:
- verifyAccountPasswordV2 / changeDatabasePasswordV2 (change password)
- verifyDatabasePasswordV2 (no usage)
- deleteMultiaccountV2 (delete profile)
- convertToKeycardAccountV2 (no usage)
- multiformatDeserializePublicKeyV2 (add contact via scanning contact QR code)
- sendTransactionV2 / encodeTransferV2 / encodeFunctionCallV2 (send transaction)
- inputConnectionStringForBootstrappingV2 / inputConnectionStringForImportingKeypairsKeystoresV2 (local sync/pairing)
- connectionChangeV2 (Offline/Online)
- appStateChangeV2 (App to background/foreground, user profile image should display normally)
- deleteImportedKeyV2 (no usage)
- migrateKeyStoreDirV2 (login with keycard)
- validateMnemonicV2 (recover from seed phrase)

For QA testing, pls ignore no usage endpoints like `verifyDatabasePasswordV2`, we should ensure the relate feature which mentioned in brackets works as before. Also we should ensure there's no sensitive information like `password`/`mnemonic` logged in `requests.log`

status: ready.

***NOTE:*** Hope someone know OC could take a deep review as the OC code was generated by AI cc @siddarthkay

### CODE REVIEW

**Comment by siddarthkay** [comment]
For objective-c code review I'd also tag @briansztamfater 
from my perspective if it works I don't have much of a problem in coding style so as long as this PR passes testing by QA I am good :D 

I'll also review later in the day, Thanks for tagging @qfrank

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m [hunk 1]
```diff
@@ -166,7 +168,20 @@ - (void)handleSignal:(NSString *)signal
...
-    StatusgoConnectionChange(type, isExpensive ? 1 : 0);
+    NSDictionary *params = @{
+        @"type": type,
+        @"expensive": isExpensive ? @YES : @NO
+    };
+    NSError *error;
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:params 
+                                                       options:0
+                                                         error:&error];
+    if (!jsonData) {
```
Comment (briansztamfater) [inline modules/react-native-status/ios/RCTStatus/RCTStatus.m]: Maybe you can check for `error` variable here

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m [hunk 2]
```diff
@@ -166,7 +168,20 @@ - (void)handleSignal:(NSString *)signal
 #if DEBUG
     NSLog(@"ConnectionChange() method called");
 #endif
-    StatusgoConnectionChange(type, isExpensive ? 1 : 0);
+    NSDictionary *params = @{
+        @"type": type,
+        @"expensive": isExpensive ? @YES : @NO
```
Comment (briansztamfater) [inline modules/react-native-status/ios/RCTStatus/RCTStatus.m]: Maybe just `@(isExpensive)` instead of using the ternary operator

**Review by briansztamfater — APPROVED** [review]
Thanks for tagging / requesting review @siddarthkay @qfrank!

From the Objective-C side looks good to me, I only added two minor comments. Seems like the new endpoint requires the parameters to be sent as a json string, and although I haven't written Objective-C code for some time, the approach looks correct. Let me know if you need help debugging it.

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m (hunk 2 above)
Comment (qfrank) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.m]: fixed

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m (hunk 1 above)
Comment (qfrank) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.m]: fixed

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/DatabaseManager.kt [hunk 3]
```diff
@@ -23,16 +25,35 @@ class DatabaseManager(private val reactContext: ReactApplicationContext) : React
...
+            val jsonParams = params.toString()
+            
+            val result = Statusgo.exportUnencryptedDatabaseV2(jsonParams)
+            if (result.startsWith("{\"error\":\"\"")) {
+                Log.d(TAG, "Export result: $result")
+            } else {
+                Log.e(TAG, "Export failed: $result")
+            }
+
+            callback.invoke(newFile.absolutePath)
```
Comment (smohamedjavid) [inline modules/react-native-status/android/src/main/java/im/status/ethereum/module/DatabaseManager.kt:55]: Quick question: If the DB export fails, should we still need to invoke the callback?

File: modules/react-native-status/ios/RCTStatus/Utils.m [hunk 4]
```diff
@@ -110,9 +131,18 @@ + (void) migrateKeystore:(NSString *)accountData
...
-        NSLog(@"validateMnemonic() method called");
+        NSLog(@"validateMnemonicV2() method called");
     #endif
-    NSString *result = StatusgoValidateMnemonic(seed);
+    NSDictionary *params = @{@"mnemonic": seed};
+    NSError *error;
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:params options:0 error:&error];
+    if (error) {
+        NSLog(@"Error creating JSON: %@", error);
+        callback(@[@"Error creating JSON"]);
```
Comment (smohamedjavid) [inline modules/react-native-status/ios/RCTStatus/Utils.m]: I'm not sure if we need to invoke the callback with a string. The native module promise expects a map in return to check if there is any error.

https://github.com/status-im/status-mobile/blob/020d5cc1c124f9bd2e8931d5275102a7b093d573/src/native_module/utils.cljs#L10

cc @clauxx

**Review by smohamedjavid — COMMENTED** [review]
Nice work! @qfrank

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/DatabaseManager.kt (hunk 3 above)
Comment (qfrank) [reply modules/react-native-status/android/src/main/java/im/status/ethereum/module/DatabaseManager.kt:55]: removed the callback invocation when export fails

File: modules/react-native-status/ios/RCTStatus/Utils.m (hunk 4 above)
Comment (qfrank) [reply modules/react-native-status/ios/RCTStatus/Utils.m]: removed all the callback invocation when the json error happen, just log the error

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt [hunk 5]
```diff
@@ -224,16 +224,25 @@ class AccountManager(private val reactContext: ReactApplicationContext) : ReactC
...
         val newKeystoreDir = utils.pathCombine(absRootDirPath, "keystore")
 
+        val jsonParams = JSONObject()
+        jsonParams.put("keyStoreDir", newKeystoreDir)
+        jsonParams.put("address", address) 
+        jsonParams.put("password", password)
+
         utils.executeRunnableStatusGoMethod(
-            { Statusgo.verifyAccountPassword(newKeystoreDir, address, password) },
+            { Statusgo.verifyAccountPasswordV2(jsonParams.toString()) },
```
Comment (siddarthkay) [inline modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt:233]: Any reason for creating a json object then adding key value pairs to it as params and then converting it to a string while calling statusgo?

to me it seems simpler to just pass in the string parameters.

```suggestion
            { Statusgo.verifyAccountPasswordV2(newKeystoreDir, address, password) },
```

Comment (siddarthkay) [inline modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt]: I couldn't find `verifyAccountPasswordV2` in status-go changes

Comment (siddarthkay) [reply modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt]: I did however find `VerifyDatabasePasswordV2`

**Comment by qfrank** [comment]
> I couldn't find verifyAccountPasswordV2 in status-go changes

[here](https://github.com/status-im/status-go/blob/dcbd854dc7fa57922492630ad88e69830a3cd628/mobile/status.go#L291) @siddarthkay

**Review by siddarthkay — APPROVED** [review]

**Comment by siddarthkay** [comment]
> > I couldn't find verifyAccountPasswordV2 in status-go changes
> 
> [here](https://github.com/status-im/status-go/blob/dcbd854dc7fa57922492630ad88e69830a3cd628/mobile/status.go#L291) @siddarthkay

my bad, when I searched your PR, initially diffs for large file were not rendered by Github UI, hence search returned no results

**Review by ilmotta — APPROVED** [review]

**Comment by pavloburykh** [comment]
@qfrank thank you for the PR. Could you please summarise what exactly should be QAed in this PR? I have read comments/description of previous PRs and still not sure I understand correctly. Thanks.

**Comment by qfrank** [comment]
> @qfrank thank you for the PR. Could you please summarise what exactly should be QAed in this PR? I have read comments/description of previous PRs and still not sure I understand correctly. Thanks.

Thank you for taking this, updated the PR description :) @pavloburykh

**Comment by pavloburykh** [comment]
Hey @qfrank thanks for the PR. Please take a look at the issue. 

### ISSUE 1 `Cannot unmarshal string into Go struct...` error when scanning Sync code by IOS

Reproducing only on IOS. iPhone 15, IOS 17.6.1

**Steps:** 

1. Generate Sync QR on Devce A
2. Scan Sync QR by Device B (IOS)
3. Observe the error

**Actual result:** `Cannot unmarshal string into Go struct...` error. Cannot proceed with sync. 

[logs (51).zip](https://github.com/user-attachments/files/17493723/logs.51.zip)
photo_2024-10-23 17 46 27

**Comment by ilmotta** [comment]
Hey @qfrank @pavloburykh even if this PR is approved for merging, let's wait until the QA team has tested the release build. Once we get the preliminary results we can re-check and see if it's okay to merge.

This is a simple strategy to reduce the risk and to avoid rebase hell in release branches. Large and/or important refactors that can affect the whole app should ideally wait until the release testing process is approaching completion/stability.

I hope that's a sensible thing to ask 🙏🏼 Thank you

**Comment by ilmotta** [comment]
FYI @qfrank, just a few more days of waiting and if nothing outstanding appears from the release testing, this PR can be merged with peace of mind 🕊️ Maybe this Friday, Nov 01 we will consider mobile 2.31 tested and ready to be shipped (we won't release yet due to other processes, that will be for next week).

**Comment by qfrank** [comment]
> Hey @qfrank thanks for the PR. Please take a look at the issue.
> 
> ### ISSUE 1 `Cannot unmarshal string into Go struct...` error when scanning Sync code by IOS
> Reproducing only on IOS. iPhone 15, IOS 17.6.1
> 
> **Steps:**
> 
> 1. Generate Sync QR on Devce A
> 2. Scan Sync QR by Device B (IOS)
> 3. Observe the error
> 
> **Actual result:** `Cannot unmarshal string into Go struct...` error. Cannot proceed with sync.
> 
> [logs (51).zip](https://github.com/user-attachments/files/17493723/logs.51.zip) photo_2024-10-23 17 46 27

issue 1 should be fixed

**Comment by pavloburykh** [comment]
> issue 1 should be fixed

Thanks @qfrank issue 1 is fixed. As we have agreed, will wait for https://github.com/status-im/status-mobile/pull/21550 and then proceed with testing of this PR.

**Comment by qfrank** [comment]
Hi @pavloburykh , the PR is ready to do manual QA now, we need to have a full regression test since this PR touch all of the function in `native_module/core.cljs` , it should work as it's on branch `develop` 🙏

**Comment by pavloburykh** [comment]
@qfrank thank you for your work. I haven't managed to find any new issue. PR is ready for merge.

**Pull request merged**